src/wormhole/session/tile/plant/nickname.rs

```diff
@@ -0,0 +1,151 @@
+use super::SessSend;
+use hcor::{id, plant, Item, ItemId, Plant, TileId};
+use log::*;
+use std::fmt;
+
+#[derive(Debug)]
+pub enum Error {
+    NoSuch(id::NoSuch),
+}
+use Error::*;
+
+impl From<id::NoSuch> for Error {
+    fn from(ns: id::NoSuch) -> Error {
+        Error::NoSuch(ns)
+    }
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "couldn't rename plant: ")?;
+        match self {
+            NoSuch(ns) => write!(f, "{}", ns),
+        }
+    }
+}
+
+pub fn nickname(ss: &mut SessSend, tile_id: TileId, new_name: String) -> Result<String, Error> {
+    let plant = ss.plant_mut(tile_id)?;
+    plant.nickname = new_name.clone();
+
+    Ok(new_name.clone())
+}
+
+#[cfg(all(feature = "hcor_client", test))]
+mod test {
+    #[actix_rt::test]
+    /// NOTE: relies on item/spawn!
+    async fn summon() -> hcor::ClientResult<()> {
+        use hcor::{Hackstead, Item, Tile};
+        use log::*;
+
+        // attempt to establish logging, do nothing if it fails
+        // (it probably fails because it's already been established in another test)
+        drop(pretty_env_logger::try_init());
+
+        let (_, seed_arch) = hcor::CONFIG
+            .seeds()
+            .next()
+            .expect("no items in config that are seeds?");
+        let not_seed_arch = hcor::CONFIG
+            .possession_archetypes
+            .iter()
+            .find(|a| a.seed.is_none())
+            .expect("no items in config that aren't seeds?");
+
+        // create bob's stead!
+        let mut bobstead = Hackstead::register().await?;
+
+        let seed_item = seed_arch.spawn().await?;
+        let not_seed_item = not_seed_arch.spawn().await?;
+        let open_tile = bobstead.free_tile().expect("fresh hackstead no open land?");
+
+        struct NewPlantAssumptions {
```

Comment:
I'm going to go out on a *limb* here and say this test doesn't test renaming plants, it tests spawning them ...

Reply:
I would suggest probably going off of i.e. rub_effects tests here, since those just panic if planting a plant goes wrong, whereas these go across every case, sometimes expecting errors, etc.